internal/hierarchyconfig/reconciler.go

```diff
@@ -846,6 +855,10 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, maxReconciles int) error
 				Name:      api.Singleton,
 				Namespace: obj.GetNamespace(),
 			}},
+			{NamespacedName: types.NamespacedName{
```

Comment:
Update the comment on line 851? And maybe add comments about why both are needed:
* The owner of the anchor needs to be reconciled to add or remove finalizers (IIRC)
* The subnamespace created by the anchor needs to be reconciled in case managed labels or annotations have changed